Adding trouble shooting docs (#21)
radicle-dev/radicle-docs: docs/trouble-shooting.md

---
id: trouble-shooting

> title: Trouble Shooting

Reviewer: "Troubleshooting" is one word.

---

> “If you know the enemy and know yourself, you need not fear the result of a
> hundred battles. If you know yourself but not the enemy, for every victory
> gained you will also suffer a defeat. If you know neither the enemy nor
> yourself, you will succumb in every battle.”
>
> ― Sun Tzu, The Art of War

As developers of Upstream we want you, the user, to have the best experience as
possible. Also as developers of Upstream, we know that there will be bugs and
errors. To help consolidate these diametrically opposed pieces of knowledge we
will try and document any of the errors that we are aware of, and give you
advice on how you might recover. In the event that this information doesn't

> help, do not hesitate to reach out to [radicle.community](community) — or our

Reviewer 2: ```suggestion
help, do not hesitate to reach out to [radicle.community][community] — or our
```

The same fix needs to be applied to the other links.

Reviewer: https://github.com/radicle-dev/radicle-docs/pull/21/commits/e7b7b2f4163fcb8932e68db1dfc256b6a30699da

> ---

Reviewer: Fixed in: https://github.com/radicle-dev/radicle-docs/pull/21/commits/4f69e2a10e0edd7040fe7f3a72590f93a7a4c21e.

id: trouble-shooting
title: Trouble Shooting
---

> “If you know the enemy and know yourself, you need not fear the result of a
> hundred battles. If you know yourself but not the enemy, for every victory
> gained you will also suffer a defeat. If you know neither the enemy nor
> yourself, you will succumb in every battle.”
>
> ― Sun Tzu, The Art of War

As developers of Upstream we want you, the user, to have the best experience as
possible. Also as developers of Upstream, we know that there will be bugs and
errors. To help consolidate these diametrically opposed pieces of knowledge we
will try and document any of the errors that we are aware of, and give you
advice on how you might recover. In the event that this information doesn't
help, do not hesitate to reach out to [radicle.community](community) — or our
[GitHub issues](github-issues) if you are comfortable with opening a detailed
issue.

> For our known failures, we categorise them with a camel-case code word that

Reviewer 3: ```suggestion
For our known failures, we categorise them with a PascalCase code word that
```

Either that or upper camel case.

`signed-refs` are the git references that peers advertise which they have also
signed with their secret key. This means that when your peer receives
`signed-refs` it attempts to verify the signature using the advertised [Device
ID](device-id). If this verification fails, then this project may be left in a
broken state.

#### Action

Some of these cases may not be errors and the `signed-refs` did not actually
come from that Device ID. If you think this is indeed an error, then we can
attempt to inspect the `signed-refs`.

To get the JSON data for the `signed-refs` we must run the following commands,
filling in the details for your particular project and remote.

```
$ cd $XDG_DATA_HOME/radicle/git

$ git ls-tree refs/namespaces/<id>/refs[/remotes/<peer>]/rad/signed_refs
100644 blob e3f6e6bd25955802060698f5b5449c874969aa71    refs

$ git cat-file -p e3f6e6bd25955802060698f5b5449c874969aa71
{"refs":{"heads":{},"remotes":{}},"signature":"hyydp6mkwkrx1cuz8i4pg3wfojujxaa3s81higa846msaroegcdspp3qe358wmpa3f85s6zmyha5zfgtqnkszwcc6a3k7pfd35fnceoec"}
```

If you report this JSON payload and the offending Device ID, we can take a look
and see if we notice anything out of the ordinary.

> ## UnexpectedBackendExit

Reviewer 2: I’ve renamed “proxy” to “backend”. I think the latter is much easier to understand for users. I’m also thinking about renaming this error to `BackendTerminated` which would make it more aligned with the usual terminology. Any thoughts

Of course this needs to be followed up with a change to the app.

### Reason

The app runs a backend process in the background that handles networking and
file system access. The error indicates that this process was either killed by
the user or the system or crashed.

### Action

The first thing you can do is try to restart the app which will restart the
backend process. If the problem persists, the logs from the process might tell
you what the issue is and give you an idea how you might fix it. For example,
another process may be listening on the port we’re using or the backend does not
have the permissions to write to a certain directory.

If you’re not able to fix the problem, please open an [issue on
GitHub][github-issues].

## UnknownException
### Reason

This is catch all error. As we categorise our errors more & more — and even
better, eliminate them! — then we should see less & less of this particular
case.

### Action

Please report these errors to [GitHub](github-issues) with as much detail as
possible — checking first that it's not already reported by another user.

> [community]: https://radicle.community/

Reviewer 3: ```suggestion
[community]: https://matrix.radicle.community/
```
